src/pages/office/components/fault-comment/fault-comment.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Input, Output, EventEmitter, OnChanges } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { FaultComment } from './fault-comment.model';
+import { SeriousFaultsContainer, FaultCount } from '../../../../shared/constants/competencies/catb-competencies';
+
+@Component({
+  selector: 'fault-comment',
+  templateUrl: 'fault-comment.html',
+})
+export class FaultCommentComponent implements OnChanges {
+
+  @Input()
+  parentForm: FormGroup;
+
+  @Input()
+  faultComment: SeriousFaultsContainer | FaultCount;
+
+  @Input()
+  faultType: string;
+
+  @Output()
+  faultCommentChange = new EventEmitter<FaultComment>();
+
+  private formControl: FormControl;
+
+  ngOnChanges(): void {
+    if (!this.formControl) {
+      this.formControl = new FormControl(null, Validators.required);
+      this.parentForm.addControl(this.formControlName, this.formControl);
+    }
+    this.formControl.patchValue(this.faultComment.comment);
+  }
+
+  faultCommentChanged(comment: string): void {
+    const faultComment: FaultComment = { comment, competency: this.faultComment.propertyName };
+    this.faultCommentChange.emit(faultComment);
+  }
+
+  get invalid() {
```

Comment:
`isValid` getter would sound better and maybe negate where ever it's need to be checked if is invalid

Reply:
I agree, however, this would mean changing it in the rest of the components too. Seeing as this is working and fulfilling the user need as it is, I will look into doing this in another PR with some other tweaks.